app/src/test/java/com/kickstarter/viewmodels/CreatorBioViewModelTest.kt

```diff
@@ -0,0 +1,86 @@
+package com.kickstarter.viewmodels
+
+import android.content.Intent
+import com.kickstarter.KSRobolectricTestCase
+import com.kickstarter.libs.Environment
+import com.kickstarter.libs.MockCurrentUser
+import com.kickstarter.mock.factories.ProjectFactory
+import com.kickstarter.mock.factories.UserFactory
+import com.kickstarter.models.Project
+import com.kickstarter.ui.IntentKey
+import org.junit.Test
+import rx.observers.TestSubscriber
+
+class CreatorBioViewModelTest : KSRobolectricTestCase() {
+    private lateinit var vm: CreatorBioViewModel.ViewModel
+
+    private val messageIconIsGone = TestSubscriber<Boolean>()
+    private val startComposeMessageActivity = TestSubscriber<Project>()
+    private val startMessagesActivity = TestSubscriber<Project>()
+    private val url = TestSubscriber<String>()
+
+    private fun setUpEnvironment(environment: Environment, project: Project? = ProjectFactory.project()) {
+        this.vm = CreatorBioViewModel.ViewModel(environment)
+
+        this.vm.outputs.messageIconIsGone().subscribe(this.messageIconIsGone)
+        this.vm.outputs.startComposeMessageActivity().subscribe(this.startComposeMessageActivity)
+        this.vm.outputs.startMessagesActivity().subscribe(this.startMessagesActivity)
+        this.vm.outputs.url().subscribe(this.url)
+
+        // Configure the view model with a project and url intent.
+        this.vm.intent(Intent().putExtra(IntentKey.PROJECT, project)
+                .putExtra(IntentKey.URL, "http://www.project.com/creator-bio"))
+    }
+
+    @Test
+    fun testLoggedOutUser() {
```

Comment:
We could rename this to `testMessageIconIsGone_WhenUserIsLoggedOut` to be more descriptive about what we're really testing.